python/dex/interop/jax/apply.py

```diff
@@ -192,16 +192,39 @@ def dex_call_batched(batched_args, batched_dims, func_atom):
   func_name = func_atom.name
   assert func_name is not None
 
+  # TODO: Make it possible to get the signature without compiling the function
+  native = get_compiled(func_atom)
+  expl_args = native.explicit_argument_signature
+  if len(batched_args) != len(expl_args):
+    raise RuntimeError(f"Dex function expects {len(expl_args)} arguments, but was given {len(batched_args)}")
+
+  def batch_type_annotation(param_idx, binder):
+    if batched_dims[param_idx] is batching.not_mapped:
+      return binder.type.dex_annotation()
+    else:
+      return f"(Fin {batch_size}) => {binder.type.dex_annotation()}"
+  batched_explicit_args = {}
+  for param_idx, binder in enumerate(expl_args):
+    ty = batch_type_annotation(param_idx, binder)
+    batched_explicit_args[binder.name] = f"({binder.name} : {ty})"
```

Comment:
Is there any reaosn why we can't inline this loop below? explicit_argument_signature is just argument_signature with implicit args filtered out. But you do already have a conditional on those in the loop below.

Reply:
If the issue is that you have to zip `batched_dims` with `explicit_params`, you can just do
`batched_dims_it = iter(batched_dims)` and then consume `next(batched_dims_it)` inside the loop. That gets rid of the extra map indirection and is IMO much easier to understand.

Reply:
Yeah, zipping with batched_dims was the issue.  Thanks, that's a reasonably clean way to handle it.